packages/webapp/src/pages/index.tsx

```diff
@@ -70,11 +79,16 @@ const HomePage: FC = wrap(function Home() {
 		(values: any) => {
 			switch (newFormName) {
 				case 'addRequestForm':
-					addEngagement(values)
+					if (isOffline && isDurableCacheEnabled) {
+						saveQueuedData(values)
+						addEngagement(values)
+					} else {
+						addEngagement(values)
```

Comment:
You don't need the `else` here and just keep one `addEngagement(values)` after your `if`

Reply:
Yeah, nice catch.